sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/ToString.java

```diff
@@ -18,51 +18,176 @@
 
 package org.apache.beam.sdk.transforms;
 
+import java.util.Iterator;
+
+import org.apache.beam.sdk.values.KV;
 import org.apache.beam.sdk.values.PCollection;
 
 /**
- * {@link PTransform PTransforms} for converting a {@link PCollection PCollection&lt;T&gt;} to a
- * {@link PCollection PCollection&lt;String&gt;}.
- *
- * <p>Example of use:
- * <pre> {@code
- * PCollection<Long> longs = ...;
- * PCollection<String> strings = longs.apply(ToString.<Long>element());
- * } </pre>
- *
+ * {@link PTransform PTransforms} for converting a {@link PCollection PCollection&lt;?&gt;},
+ * {@link PCollection PCollection&lt;KV&lt;?,?&gt;&gt;}, or
+ * {@link PCollection PCollection&lt;Iterable&lt;?&gt;&gt;}
+ * to a {@link PCollection PCollection&lt;String&gt;}.
  *
  * <p><b>Note</b>: For any custom string conversion and formatting, we recommend applying your own
  * {@link SerializableFunction} using {@link MapElements#via(SerializableFunction)}
  */
 public final class ToString {
+  private ToString() {
+    // do not instantiate
+  }
 
   /**
    * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
    * element of the input {@link PCollection} to a {@link String} using the
    * {@link Object#toString} method.
    */
-  public static PTransform<PCollection<?>, PCollection<String>> element() {
-    return new Default();
+  public static PTransform<PCollection<?>, PCollection<String>> of() {
+    return new SimpleToString();
   }
 
-  private ToString() {
+  /**
+   * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
+   * element of the input {@link PCollection} to a {@link String} using the
+   * {@link Object#toString} method.
+   */
+  public static PTransform<PCollection<? extends KV<?, ?>>, PCollection<String>> kv() {
+    return kv(",");
+  }
+
+  /**
+   * Returns a {@code PTransform<PCollection, PCollection<String>>} which transforms each
+   * element of the input {@link PCollection} to a {@link String} using the
+   * {@link Object#toString} method.
```

Comment:
Other classes have the JavaDoc here. I think we need to do a general pass and make them all consistent.